app/src/main/java/com/pps/globant/fittracker/mvp/view/WeatherView.java

```diff
@@ -0,0 +1,141 @@
+    TextView weatherTextCloudinessValue;
+    @BindView(R.id.weather_text_humidity_value)
+    TextView weatherTextHumidityValue;
+    @BindView(R.id.weather_text_pressure_value)
+    TextView weatherTextPressureValue;
+    @BindView(R.id.weather_text_sunrise_value)
+    TextView weatherTextSunriseValue;
+    @BindView(R.id.weather_text_sunset_value)
+    TextView weatherTextSunsetValue;
+    @BindView(R.id.weather_text_rain_value)
+    TextView weatherTextRainValue;
+    @BindView(R.id.button_weather_see_more)
+    Button buttonSeeMore;
+    @BindView(R.id.weather_progress)
+    ProgressBar weatherProgress;
+
+
+    public WeatherView(CardView cardView, Activity activity, Bus bus) {
+        super(cardView, activity);
+        ButterKnife.bind(this, activity);
+        this.bus = bus;
+    }
+
+    public void requestPermission(Activity activity, String permission, int requestCode) {
+        ActivityCompat.requestPermissions(activity, new String[]{permission}, requestCode);
+    }
+
+    @OnClick(R.id.button_weather_permission)
+    public void weatherPermissionButtonPressed() {
+        bus.post(new WeatherPermissionButtonPressedEvent());
+    }
+
+    @OnClick(R.id.button_weather_refresh)
+    public void weatherRefreshButtonPressed() {
+        bus.post(new WeatherRefreshButtonPressedEvent());
+    }
+
+    @OnClick(R.id.button_weather_see_more)
+    public void weatherSeeMoreButtonPressed() {
+        bus.post(new WeatherSeeMoreButtonPressedEvent());
+    }
+
+    public void showExtendedWeather(double speed, String direction, int humidity, double pressure, int cloud, double rain, String sunrise, String sunset) {
+        if (getContext() != null) {
+            weatherTextWindSpeedValue.setText(String.format(getContext().getResources().getString(R.string.wind_speed_value), speed));
+            weatherTextHumidityValue.setText(String.format(getContext().getResources().getString(R.string.humidity_value), humidity));
+            weatherTextPressureValue.setText(String.format(getContext().getResources().getString(R.string.pressure_value), pressure));
+            weatherTextCloudinessValue.setText(String.format(getContext().getResources().getString(R.string.cloudiness_value), cloud));
+            weatherTextRainValue.setText(String.format(getContext().getResources().getString(R.string.rain_value), rain));
+            weatherTextSunriseValue.setText(sunrise);
+            weatherTextSunsetValue.setText(sunset);
+            weatherTextWindDirectionValue.setText(direction);
+            buttonSeeMore.setText(getContext().getResources().getString(R.string.cards_see_less));
+            weatherLayoutExpanded.setVisibility(View.VISIBLE);
+        }
+    }
+
+    public void hideExtended() {
+        if (getContext() != null) {
+            buttonSeeMore.setText(getContext().getResources().getString(R.string.cards_see_more));
+            weatherLayoutExpanded.setVisibility(View.GONE);
+        }
+    }
+
+    public void toggleSpinner() {
```

Comment:
split this method in 2(toggle on/off). we don't have to add ANY logic in the view.